tribe-ext-only-city-based-timezones.php

```diff
@@ -0,0 +1,170 @@
+ * Text Domain:     tribe-ext-only-city-based-timezones
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if (
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( 'Tribe__Extension__Allow_Only_City_Based_Timezones' )
+) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Allow_Only_City_Based_Timezones extends Tribe__Extension {
+		/**
+		 * Setup the Extension's properties.
+		 *
+		 * This always executes even if the required plugins are not present.
+		 */
+		public function construct() {
+			// readme.txt states WP 4.7.0+ is required because that's when wp_timezone_choice() added the `$locale` parameter
+			// tribe_events_timezone_choice() didn't exist until TEC 4.6.5
+			$this->add_required_plugin( 'Tribe__Events__Main', '4.6.5' );
+			$this->set_url( 'https://theeventscalendar.com/extensions/allow-only-city-based-timezones-for-events/' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-only-city-based-timezones', false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			add_filter( 'tribe_events_timezone_choice', array( $this, 'wp_timezone_choice_wo_manual_offsets' ), 10, 3 );
+		}
+
+		/**
+		 * Build the <optgroup> list of timezone choices, including "UTC" and
+		 * all city-based timezones, excluding all manual offsets like "UTC+1".
+		 *
+		 * This is a copy of wp_timezone_choice() with the UTC/GMT manual offset part of
+		 * it removed and our own text domain added.
+		 *
+		 * @see \wp_timezone_choice()
+		 * @see \tribe_events_timezone_choice()
+		 *
+		 * @return string
+		 */
+		public function wp_timezone_choice_wo_manual_offsets( $timezone_choices, $selected_zone, $locale ) {
+			static $mo_loaded = false, $locale_loaded = null;
+
+			$continents = array( 'Africa', 'America', 'Antarctica', 'Arctic', 'Asia', 'Atlantic', 'Australia', 'Europe', 'Indian', 'Pacific');
+
+			// Load translations for continents and cities.
+			if ( ! $mo_loaded || $locale !== $locale_loaded ) {
+				$locale_loaded = $locale ? $locale : get_locale();
+				$mofile = WP_LANG_DIR . '/continents-cities-' . $locale_loaded . '.mo';
+				unload_textdomain( 'continents-cities' );
+				load_textdomain( 'continents-cities', $mofile );
+				$mo_loaded = true;
+			}
+
+			$zonen = array();
```

Comment:
Why `$zonen` instead of `$zones`?

Reply:
just copied straight from wp_timezone_choice() - will fix though